lib/jarvis/defer.rb

```diff
@@ -0,0 +1,15 @@
+module Jarvis class Defer
+  def initialize
+    @deferred = []
+  end
+
+  def do(&block)
+    @deferred << block
+  end
+
+  def run
+    @deferred.each do |block|
+      block.call
+    end
+  end
+end end
```

Comment:
You mean having run() just schedule some futures instead of calling each individually? I like that idea.